File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    dic = {}
#   keymap에서 각 알파벳이 존재하는 최소 index dictionary 생성
    for key in keymap:
        for i in range(len(key)):
            if key[i] in dic:
                if dic[key[i]] > i+1:
                    dic[key[i]] = i+1
                else:
                    continue
            else:
                dic[key[i]] = i+1
#   targets 값 찾기
    for target in targets:
        tmp = 0
        for i in range(len(target)):
            if target[i] not in dic:
                tmp = -1
                break
            else:
                tmp += dic[target[i]]
        answer.append(tmp)
    return answer

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
    var answer = [];
    const dic = {};
    
    for(key in keymap){
        for(let i = 0; i < keymap[key].length; i++){
            if(keymap[key][i] in dic){
                if(dic[keymap[key][i]] > i){
                    dic[keymap[key][i]] = i + 1
                }
            } else{
                dic[keymap[key][i]] = i + 1
            }
        }
    }
    for (target of targets){
        let tmp = 0;
        for (t in target){
            tmp += dic[target[t]]
        }
        if (tmp > 0){
            answer.push(tmp)            
        } else{
            answer.push(-1)
        }
    }
    return answer;
}